### PR TITLE
Wip/gui integration2

### PR DESCRIPTION
**Pull request by tobiasschweizer**
ongoing integration of the old GUI into Knora:

- add version handling to `propedit.js`: after each value change, the new valueIri has to be stored
- values have to be submitted providing the correct type: `int_val` etc.
- there is only richtext, no simple text any more
- fix bugs in date jQuery-plugin
- fix bug in link update
- add Selenium tests (not complete yet)
- add full quality file paths to context queries
- optimize context query
- refactor redundancy checking methods for value and version creation

### CODE REVIEW

**Comment by benjamingeer** [comment]
Please add some RST documentation with instructions on how to set up Sipi to work with Knora, and anything that needs to be configured so I can test this.

**Comment by tobiasschweizer** [comment]
I added some RST documentation about how to set up Sipi and run it with Knora.

I also merged develop into this branch and resolved a conflict in the `.sbt`-files.

**Comment by benjamingeer** [comment]
Please improve the procedure for setting up Sipi to serve a dummy image. Instead of having to modify the source code, it would be nice if I could just run Sipi with a different config file provided in Sipi's git repository. Then it could serve a dummy image already provided in the git repository. Also, the documentation says that the image has to be in `imgroot`, but actually it has to be in `imgroot/knora`, otherwise you get a 404.

**Comment by benjamingeer** [comment]
The tests in `salsah` require `salsah/lib/chromedriver`. This should be documented, and if the driver isn't installed, the user should get a friendly exception telling them where to download it from and where to install it.

**Comment by tobiasschweizer** [comment]
> Also, the documentation says that the image has to be in imgroot, but actually it has to be in imgroot/knora, otherwise you get a 404.

This is related to the setting `prefix_as_path` in the config file:

```lua
 --
 -- If FALSE, the prefix is not used to build the path to the image files
 --
 prefix_as_path = true,

```

**Comment by tobiasschweizer** [comment]
Ok, I added the test config files to Sipi and assigned the pull request to you.

I added some instruction to RST how to run Sipi in test mode.

**Comment by tobiasschweizer** [comment]
I added a check for Chrome Driver and added instructions to the docs where to get it.

File: docs/rst/salsah/development/build-process.rst [hunk 1]
```diff
@@ -64,21 +64,25 @@ Then in another terminal, go to the SALSAH root directory and start the server:
...
   > re-stop
 
 
 Run the automated tests
 ------------------------
 
-Make sure you've started Fuseki and the API server as shown above. Then at the SBT prompt:
+Make sure you've started Fuseki and the API server as shown above.
+In order to run the tests, the Selenium driver for Chrome has to be installed.
+Please download it from https://sites.google.com/a/chromium.org/chromedriver/downloads and save it as ``salsah/lib/chromedriver``.
```
Comment (benjamingeer) [inline docs/rst/salsah/development/build-process.rst]: Please make this a link in RST.

File: salsah/src/test/scala/org.knora.salsah/browser/SalsahPage.scala [hunk 2]
```diff
@@ -41,6 +46,12 @@ import org.scalatest.concurrent.Eventually._
...
     val pageUrl = "http://localhost:3335/index.html" // TODO: get this from application.conf
 
+    val chromeDriverPath = "lib/chromedriver"
+
+    // check if the Selenium driver for Chrome exists
+    val SeleniumChromeDriver = new File(chromeDriverPath)
+    if (!SeleniumChromeDriver.exists) throw new FileNotFoundException(s"${chromeDriverPath} could not be found. Please install Selenium Chrome Driver.")
+
     // Load the native Selenium driver for Chrome.
     System.setProperty("webdriver.chrome.driver", "lib/chromedriver")
```
Comment (benjamingeer) [inline salsah/src/test/scala/org.knora.salsah/browser/SalsahPage.scala]: Please use the variable `chromeDriverPath` here.

File: salsah/src/test/scala/org.knora.salsah/browser/SalsahPage.scala [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    val pageUrl = "http://localhost:3335/index.html" // TODO: get this from application.conf
+
+    val chromeDriverPath = "lib/chromedriver"
+
+    // check if the Selenium driver for Chrome exists
+    val SeleniumChromeDriver = new File(chromeDriverPath)
+    if (!SeleniumChromeDriver.exists) throw new FileNotFoundException(s"${chromeDriverPath} could not be found. Please install Selenium Chrome Driver.")
+
+    // Load the native Selenium driver for Chrome.
+    System.setProperty("webdriver.chrome.driver", "lib/chromedriver")
```
Comment (benjamingeer) [inline salsah/src/test/scala/org.knora.salsah/browser/SalsahPage.scala]: Please use the variable `chromeDriverPath` here.

**Comment by benjamingeer** [comment]
Please just fix the two things I commented on above, and then go ahead and merge.

**Comment by subotic** [comment]
resolves #111

**Pull request merged**